xmipp

```diff
@@ -841,23 +880,41 @@ def checkConfig():
         if not checkCuda(configDict):
             print(red("Cannot compile with NVCC, continuing without CUDA"))
             runJob("rm xmipp_cuda_test*")  # if fails, the test files remains
-            configDict["CUDA"]="False"
+            newConf["CUDA"]="False"
         if not checkMatlab(configDict):
             print(red("Cannot compile with Matlab, continuing without Matlab"))
-            configDict["MATLAB"]="False"
-        configDict['VERIFIED']="True"
-        writeConfig(configDict)
+            newConf["MATLAB"]="False"
+        newConf['VERIFIED']="True"
+        updateConfig(newConf)
     return True
 
 
-def compileModule(Nproc,module):
+def compileModule(Nproc,module, tries=0):
     shutil.copyfile(CONFIG_FILE_NAME,"src/%s/install/%s" % (module, CONFIG_FILE_NAME))
     if module == "xmipp":
         stampVersion()
-    ok = runJob("scons -j%s"%Nproc, "src/%s"%module)
+    log = []
+    ok = runJob("scons -j%s"%Nproc, "src/%s"%module, log=log)
+    if not ok and tries < 6:
+        newConf = {}
+        for l in log[-30:]:  # inspecting last 30 lines
+            # expected error: 'scons: *** [some/program/to/compile] Error 1'
+            failingBin = re.match("scons: \*\*\* \[(.*)\] Error .*", l)
+            if failingBin:
+                for k, v in SKIPPABLE_BINS.iteritems():
+                    if k in failingBin.group(1):
```

Comment:
This doesn't sound right.
What use cases do we have?

- generate the config file: in that case, we should not generate any settings which is not valid
- reuse existing config file

In the later case, either we generated it (hence it should work), or user changed it, and then it's his responsibility to set it properly.

I disagree in changing the config file, as it might be very unintuitive for the user.

Reply:
I haven't see the details, but I agree very much with @DStrelak in this. The installation process should not modify the config file, either generate it correctly or warn to things the user should correct, but not changing it. Maybe it is more useful a check or validation that assist the user how to correctly change it.

Reply:
Well, we have cases where openCV is found but during the compilation it fails somehow (I have a case reported by mail from one beta-tester). With this, I try to avoid failures during the installation when the failure is not in a critical scope.

Let's imagine a user trying to install Xmipp in Scipion's plugin manager (**the vast majority of the users**). If a openCV code compilation fails somehow (linking failure or whatever not detected during the check-config), then how we can to assist to the user? Suggesting him/her to edit the software/em/xmippSrc-vNum/xmipp.conf when they doesn't know so much about that config file.. and, then, how to process to the next trying by using the edited xmipp.conf with the plugin manager?

Under my point of view, the xmipp.conf is almost transparent to the final user and it could be more a summary of the configuration than a tuning file (however, it can be used as so).

Reply:
During the plugin installation, the xmipp.conf should be set according to the one of the Scipion, is that right, @pconesa ?
I don't know if Scipion has a way to determine that the (dependency) paths are set correctly prior the installation of the plugins.

I think that proper walkthrough should be sth like this:
- either Xmipp or Scipion create a config file, provided non existed before (we should properly ignore / handle config files from previous versions) . Content should be rather safe - if we cannot check that a certain settings is working, we should ignore it.
- if there is a config file (for current version of Scipion or Xmipp), we can expect that it is
  - either generated by us (i.e. correct)
  - altered by user (in that case, it his/her problem to set it correctly, we can only verify)

I, as a Scipion user installing Xmipp, I can imagine getting a window with check boxes and path text boxes:

- [ ] I want Cuda
- Path to cuda libraries ...
- Path to cuda executables ...
- [ ] I want OpenCV
- Path to OpenCV libraries ...
- Path to OpenCV executables ...

Based on this, the Scipion's (hence eventually the Xmipp's) config file should be set/updated.
We can do some basic check that the paths are not complete rubbish, we can try to detect the paths automatically, we can detect incompatibilities, but it ends there. 
Should the user get a compilation error then, he/she will have to fix it.

Reply:
@pconesa gave me an idea. The thing is, if `OPENCV=False` is in the environ, then no assertion is needed to set that config var, just take it and that's it.

This allows to hint the user to either `export OPENCV=False` or edit the `config/scipion.conf`. (This last only if SCIPION_HOME is defined).

How it sounds?

Reply:
Ok. Now the following hint is gift if a skkipable bin is found like `movie_optical_alignment_cpu`:
> Some error compiling 'movie_optical_alignment_cpu.o' program.
   You can skip this program by including 'OPENCV=False' in the config file.

I checked both:
 1) editing `scipion.conf` and relauching the installation in the plugin manager works fine.
 2) editing `xmipp.conf` and running again `./xmipp` (or `./xmipp compile` or `./xmipp compileAndInstall`) works fine.